integration-tests/build.gradle.kts

```
@@ -21,7 +21,26 @@ subprojects {
         implementation(project(":integration-tests"))
     }
 
-    val integrationTest by tasks.register<Test>("integrationTest") {
+    /**
+     * Dokka's integration test task is not cacheable because the HTML outputs
+     * it produces when running the tests are used for showcasing resulting documentation.
+     *
+     * At the moment there are two problems that do not allow to make it cacheable:
+     * 1. The task's inputs are such that changes in Dokka's code do not invalidate the cache,
+     *    because it is run with the same version of Dokka ("DOKKA_VERSION") on the same
+     *    test project inputs.
+     * 2. The tests generate HTML output which is then used to showcase documentation.
+     *    The outputs are usually copied to a location from which it will be served.
+     *    However, if the test is cacheable, it produces no outputs, so no documentation
+     *    to showcase. It needs to be broken into two separate tasks: one cacheable for running
+     *    the tests and producing HTML output, and another non-cacheable for copying the output.
+     *
+     * @see [org.jetbrains.dokka.it.TestOutputCopier] for more details on showcasing documentation
+     */
+    @DisableCachingByDefault
+    abstract class NonCacheableIntegrationTest : Test()
+
+    val integrationTest by tasks.register<NonCacheableIntegrationTest>("integrationTest") {
```

If this implementation stays, I don't think it's necessary to create a separate task for this:
```suggestion
    val integrationTest by tasks.register<Test>("integrationTest") {
        outputs.doNotCacheIf { true }
```
see https://docs.gradle.org/current/javadoc/org/gradle/work/DisableCachingByDefault.html

---

I tried it, but it didn't work - the tests still used caches and completed in an instant.

I think it's related to the following behaviour:

> (from the Javadoc for [doNotCacheIf](https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/TaskOutputs.html#doNotCacheIf-java.lang.String-org.gradle.api.specs.Spec-))
> The spec will be evaluated at task execution time, not during configuration.

It might work if all of the current caches are reset on CI and locally, but even then I'm not sure if it's not going to cache the task itself (not the output) after running it once, because the inputs for the task are the same between runs.

---

I'm surprised that `doNotCacheIf { true }` doesn't work, but the annotation does... oh well, Gradle is weird.


I suspect that registering the environment variable as an input property would help

```kotlin
inputs.property("dokkaTestOutputPath", proiders.environmentVariable("DOKKA_TEST_OUTPUT_PATH"))
```

---

ah, that's weird, ok